api/users_test.go

```diff
@@ -378,3 +379,115 @@ func TestChangePasswordHandler(t *testing.T) {
 		}
 	})
 }
+
+func TestPasswordResetHandler(t *testing.T) {
+
+	var (
+		routeMux                                        = mux.NewRouter()
+		ctx                                             = context.Background()
+		poolId, clientId, clientSecret, authFlow string = "us-west-11_bxushuds", "abc123", "bsjahsaj9djsiq", "authflow"
+		email                                    string = "foo_bar123@ext.ons.gov.uk"
+	)
+
+	m := &mock.MockCognitoIdentityProviderClient{}
+
+	api, _ := Setup(ctx, routeMux, m, poolId, clientId, clientSecret, authFlow)
+	w := httptest.NewRecorder()
+
+	Convey("ForgotPassword - check expected responses", t, func() {
+		respondToAuthChallengeTests := []struct {
+			forgotPasswordFunction func(input *cognitoidentityprovider.ForgotPasswordInput) (*cognitoidentityprovider.ForgotPasswordOutput, error)
+			httpResponse           int
+		}{
+			{
+				// Cognito successful password change
+				func(input *cognitoidentityprovider.ForgotPasswordInput) (*cognitoidentityprovider.ForgotPasswordOutput, error) {
+					return &cognitoidentityprovider.ForgotPasswordOutput{
+						CodeDeliveryDetails: &cognitoidentityprovider.CodeDeliveryDetailsType{},
+					}, nil
+				},
+				http.StatusAccepted,
+			},
+			{
+				// Cognito internal error
+				func(input *cognitoidentityprovider.ForgotPasswordInput) (*cognitoidentityprovider.ForgotPasswordOutput, error) {
+					awsErrCode := "InternalErrorException"
+					awsErrMessage := "Something strange happened"
+					awsOrigErr := errors.New(awsErrCode)
+					awsErr := awserr.New(awsErrCode, awsErrMessage, awsOrigErr)
+					return nil, awsErr
+				},
+				http.StatusInternalServerError,
+			},
+			{
+				// Cognito invalid user
+				func(input *cognitoidentityprovider.ForgotPasswordInput) (*cognitoidentityprovider.ForgotPasswordOutput, error) {
+					awsErrCode := "CodeMismatchException"
```

Comment:
https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ForgotPassword.html- Cannot find "CodeMismatchException". Am I looking at the correct list

Reply:
I change the test scenario but missed changing the AWS Error code.
Should be 'UserNotFoundException', I've updated it now.
Tests were passing as they are both 400 errors handled in the same way in the handler.